pkg/agent/route/route_windows.go

```diff
@@ -382,13 +382,28 @@ func (c *Client) DeleteSNATRule(mark uint32) error {
 	return nil
 }
 
-// TODO: nodePortAddresses is not supported currently.
 func (c *Client) AddNodePort(nodePortAddresses []net.IP, port uint16, protocol binding.Protocol) error {
-	return util.ReplaceNetNatStaticMapping(antreaNatNodePort, "0.0.0.0", port, config.VirtualServiceIPv4.String(), port, string(protocol))
+	if len(nodePortAddresses) == 0 {
+		return util.ReplaceNetNatStaticMapping(antreaNatNodePort, "0.0.0.0", port, config.VirtualServiceIPv4.String(), port, string(protocol))
+	}
+	for _, addr := range nodePortAddresses {
+		if err := util.ReplaceNetNatStaticMapping(antreaNatNodePort, addr.String(), port, config.VirtualServiceIPv4.String(), port, string(protocol)); err != nil {
```

Comment:
In my mind, I failed to use the Node's IP or Node's transport IP in the field "ExternalIPAddress" instead of "0.0.0.0". So I think this might not work for the user provided NodePortAddresses either.

Reply:
Have you verified whether the change can work? @lzhecheng

Reply:
I am working on solving this issue. Sometimes the expected "ExternalIPAddress" doesn't appear. If not appear, I haven't find a way to "New" one. Still investigating.